File: solutions/217.Contains_Duplicate/TLE_brute_force_n2.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        TLE_brute_force_n2.cpp
*  Create Date: 2015-07-13 19:55:30
*  Descripton:   
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        for (int i = 0; i < nums.size(); ++i) {
            for (int j = i + 1; j < nums.size(); ++j) {
                if (nums[i] == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
};

int main() {

    return 0;
}

File: solutions/238.Product_of_Array_Except_Self/TLE_brute_force_n2.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Author:      illuz <iilluzen[at]gmail.com>
# File:        TLE_brute_force_n2.py
# Create Date: 2016-02-24 19:09:56
# Usage:       AC_brute_force_n.py 
# Descripton:  


class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        return [reduce(lambda x, y: x * y, nums[:i] + nums[i+1:]) for i in range(len(nums))]

s = Solution()
print(s.productExceptSelf([1, 2, 3, 4]))
